filebeat/crawler/crawler.go

```diff
@@ -6,26 +6,25 @@ import (
 
 	"github.com/elastic/beats/filebeat/input/file"
 	"github.com/elastic/beats/filebeat/prospector"
-	"github.com/elastic/beats/filebeat/spooler"
 	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/logp"
 )
 
 type Crawler struct {
 	prospectors       []*prospector.Prospector
 	prospectorConfigs []*common.Config
-	spooler           *spooler.Spooler
+	out               prospector.Outlet
```

Comment:
The problem here is, that the package name suggests its something from the prospector. Abstraction does not prevent us to add docs to make reading the code easier. In general I think abstraction is nice but often makes code harder to read. So adding docs / comments shouldn't hurt.